--- ../src-base/minecraft/net/minecraft/client/renderer/block/model/BakedQuad.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/block/model/BakedQuad.java
@@ -13,8 +13,24 @@
     protected final EnumFacing field_178214_c;
     protected final TextureAtlasSprite field_187509_d;
 
+    /**
+     * @deprecated Use constructor with the format argument.
+     */
+    @Deprecated
     public BakedQuad(int[] p_i46574_1_, int p_i46574_2_, EnumFacing p_i46574_3_, TextureAtlasSprite p_i46574_4_)
     {
+        this(p_i46574_1_, p_i46574_2_, p_i46574_3_, p_i46574_4_, true, net.minecraft.client.renderer.vertex.DefaultVertexFormats.field_176599_b);
+    }
+
+    public BakedQuad(int[] vertexDataIn, int tintIndexIn, EnumFacing faceIn, TextureAtlasSprite spriteIn, boolean applyDiffuseLighting)
+    {
+        this(vertexDataIn, tintIndexIn, faceIn, spriteIn, applyDiffuseLighting, net.minecraft.client.renderer.vertex.DefaultVertexFormats.field_176599_b);
+    }
+
+    @Deprecated
+    public BakedQuad(int[] p_i46574_1_, int p_i46574_2_, EnumFacing p_i46574_3_, TextureAtlasSprite p_i46574_4_, boolean applyDiffuseLighting, net.minecraft.client.renderer.vertex.VertexFormat format)
+    {
+        this.applyDiffuseLighting = applyDiffuseLighting;
         this.field_178215_a = p_i46574_1_;
         this.field_178213_b = p_i46574_2_;
         this.field_178214_c = p_i46574_3_;
@@ -45,4 +61,11 @@
     {
         return this.field_178214_c;
     }
+
+    protected final boolean applyDiffuseLighting;
+
+    public boolean shouldApplyDiffuseLighting()
+    {
+        return applyDiffuseLighting;
+    }
 }
